includes/utilities.inc

```diff
@@ -987,3 +990,68 @@ function islandora_paged_content_extract_tiff_from_pdf($uri, $offset, $device, $
   }
   return FALSE;
 }
+
+/**
+ * Helper to determine whether all requirements are met to extract from PDF.
+ *
+ * @return bool
+ *   TRUE if all requirements are present, FALSE otherwise.
+ */
+function islandora_paged_content_can_extract_from_pdf() {
+  return islandora_paged_content_can_combine_pdf() && islandora_paged_content_pdftotext_availability() && islandora_paged_content_pdfinfo_availability();
+}
+
+/**
+ * Helper function to add the PDF datastream to an object.
+ *
+ * @param AbstractObject $object
+ *   An AbstractObject representing an object within Fedora.
+ *
+ * @param object $file_obj
+ *   The PDF file object that is being added.
+ */
+function islandora_paged_content_add_pdf_to_object(AbstractObject $object, $file_obj) {
+  module_load_include('inc', 'islandora_paged_content', 'includes/db');
+  if (!isset($object['PDF'])) {
+    $pdf = $object->constructDatastream('PDF', 'M');
+    $object->ingestDatastream($pdf);
```

Comment:
Should move this to the bottom of the function so it won't create extra versions if the object is already ingested in the future.